### DISCUSSION TITLE
Understanding Spherical Harmonics Representing a Signal on a Sphere

### DISCUSSION THREAD

**Discussion by dangthatsright** [e3nn/e3nn#392] [Q&A]
I was looking for a function that gives you the coefficient of the spherical harmonics representing a collection of vectors.

Originally I thought that this was done via `e3nn.o3.spherical_harmonics([0,1,2], 3D_vectors)` but the output shape of this is `(num 3D vectors, 2*2+1)` so I thought that I should just sum this up since spherical harmonics are linear.

However, now I come across `e3nn.io.SphericalTensor(2,1,-1).with_peaks_at(3D_vectors)` which seems like it makes more sense for what I want.

My question is: what is the difference between these two approaches? What do these methods give me exactly? 

Edit: I also see `Sphererical.sum_of_diracts`. What is the difference between this and `with_peaks_at`? 

Thanks!

**Comment by mariogeiger** [comment]
`Sphererical.sum_of_diracts` is similar to what you did first (maybe up to some multiplicative constant?)

`with_peaks_at` is more tricky. It cancels out the influences of other peaks (due to finite lmax) by solving a linear system of equations.

A 1D example would be
```
peak1:  0    0.1    -0.1   1      -0.1   0.1    0     0      0
peak2:  0    0       0     0.1    -0.1   1     -0.1   0.1    0    

Linear system:
a1 1 + a2 0.1 = 1
a1 01 + a2 1 = 1 
```
